File: BaekJoon_10026.java
// Baekjoon 10026
// https://www.acmicpc.net/problem/10026

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.List;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int N = Integer.parseInt(br.readLine());
        String[][] image = new String[N][N];
        String[][] color_weak = new String[N][N];

        for (int n=0; n<N; n++) {
            image[n] = br.readLine().split("");
            color_weak[n] = image[n].clone();
            for (int m=0; m<N; m++)
                if (color_weak[n][m].equals("G"))
                    color_weak[n][m] = "R";
        }

        br.close();

        System.out.print(BFS(N, image));
        System.out.print(" ");
        System.out.print(BFS(N, color_weak));
    }

    static int BFS(int N, String[][] img) {
        int answer = 0;

        for (int r=0; r<N; r++)
            for (int c=0; c<N; c++) {
                if (!img[r][c].equals("N")) {
                    String target = img[r][c];
                    List<int[]> bfs = new ArrayList<>();
                    bfs.add(new int[]{r, c});

                    while (bfs.size() != 0) {
                        int[] state;
                        state = bfs.get(0);
                        bfs.remove(0);

                        List<int[]> tmp = new ArrayList<>();
                        tmp.add(new int[]{Math.max(0, state[0]-1), state[1]});
                        tmp.add(new int[]{Math.min(state[0]+1, N-1), state[1]});
                        tmp.add(new int[]{state[0], Math.max(0, state[1]-1)});
                        tmp.add(new int[]{state[0], Math.min(state[1]+1, N-1)});

                        for (int[] next: tmp) {
                            if (img[next[0]][next[1]].equals(target)) {
                                bfs.add(new int[]{next[0], next[1]});
                                img[next[0]][next[1]] = "N";
                            }
                        }
                    }
                    answer++;
                }
            }
        return answer;
    }
}

File: BaekJoon_10026.py
# Baekjoon 10026
# https://www.acmicpc.net/problem/10026

from collections import deque
from sys import stdin

rd = stdin.readline


def BFS(board, i, j):
    target = board[i][j]
    bfs = deque([(i, j)])
    while bfs:
        x_current, y_current = bfs.popleft()

        tmp = [(max(0, x_current - 1), y_current),
               (min(x_current + 1, N - 1), y_current),
               (x_current, max(0, y_current - 1)),
               (x_current, min(y_current + 1, N - 1))
               ]

        for x_next, y_next in tmp:
            if board[x_next][y_next] == target and not visited[x_next][y_next]:
                bfs.append((x_next, y_next))
                visited[x_next][y_next] = True


N = int(rd())
img = [list(rd().rstrip()) for _ in " " * N]
visited = [[False] * N for _ in " " * N]

answer1 = 0
for r, line in enumerate(img):
    for c, color in enumerate(line):
        if not visited[r][c]:
            BFS(img, r, c)
            answer1 += 1

new_img = [list("".join(row).replace("R", "G")) for row in img]
visited = [[False] * N for _ in " " * N]

answer2 = 0
for r, line in enumerate(new_img):
    for c, color in enumerate(line):
        if not visited[r][c]:
            BFS(new_img, r, c)
            answer2 += 1

print(answer1, answer2)
